yt/yt/server/http_proxy/component_discovery.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <yt/yt/core/misc/error.h>
+
+#include <yt/yt/ytlib/api/public.h>
+#include <yt/yt/ytlib/api/native/public.h>
+
+namespace NYT::NHttpProxy {
+
+////////////////////////////////////////////////////////////////////////////////
+
+//! Plural (!) names of cluster components.
```

Comment:
I just wanted to stick with either singular or plural for all components, because the fact that it's "//sys/primary_masters", but "//sys/scheduler" is annoying. I wanted to hide this in the implementation.

You are probably right that singular looks nicer. I will change it.